### PR TITLE
chore: modify new analytics utils

### PR DESCRIPTION
**Pull request by JeevaRamu0104**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [x] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
Refactored the new analytics utils

## Motivation and Context
Extract the query data and metadata in respective module
image

## How did you test it?
It can't be tested

## Where to test it?

- [ ] INTEG
- [ ] SANDBOX
- [ ] PROD

## Checklist

- [ ] I ran `npm run re:build`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: src/screens/NewAnalytics/PaymentAnalytics/NewPaymentAnalyticsUtils.res [hunk 1]
```diff
@@ -18,50 +18,57 @@ let getMonthName = month => {
...
+  ~key: string,
+  ~name: string,
+  ~color,
+): LineGraphTypes.dataObj => {
+  let data = array->Array.map(item => {
+    item->getDictFromJsonObject->getInt(key, 0)
+  })
+  let dataObj: LineGraphTypes.dataObj = {
+    showInLegend: false,
+    name: `${name}`,
```
Comment (Riddhiagrawal001) [inline src/screens/NewAnalytics/PaymentAnalytics/NewPaymentAnalyticsUtils.res:49]:
```suggestion
    name,
```

File: src/screens/NewAnalytics/PaymentAnalytics/NewPaymentAnalyticsUtils.res [hunk 2]
```diff
@@ -18,50 +18,57 @@ let getMonthName = month => {
...
+  ~key: string,
+  ~name: string,
+  ~color,
+): BarGraphTypes.dataObj => {
+  let data = array->Array.map(item => {
+    item->getDictFromJsonObject->getInt(key, 0)
   })
+  let dataObj: BarGraphTypes.dataObj = {
+    showInLegend: false,
+    name: `${name}`,
```
Comment (Riddhiagrawal001) [inline src/screens/NewAnalytics/PaymentAnalytics/NewPaymentAnalyticsUtils.res:67]:
```suggestion
    name,
```

File: src/screens/NewAnalytics/PaymentAnalytics/PaymentsProcessed/PaymentsProcessed.res [hunk 3]
```diff
@@ -16,14 +16,11 @@ module TableModule = {
...
-        ->getArrayFromDict("queryData", [])
-        ->Array.map(getDictFromJsonObject)
-        ->Array.map(tableItemToObjMapper)
-      }
-
+    | Some(val) =>
+      val
+      ->getArrayFromJson([])
+      ->Array.map(getDictFromJsonObject)
+      ->Array.map(tableItemToObjMapper)
```
Comment (Riddhiagrawal001) [inline src/screens/NewAnalytics/PaymentAnalytics/PaymentsProcessed/PaymentsProcessed.res:23]: for this is it possible to use `Logicutils.getArrayDataFromJson`

File: src/screens/NewAnalytics/PaymentAnalytics/PaymentsProcessed/PaymentsProcessedUtils.res [hunk 4]
```diff
@@ -8,14 +8,28 @@ let paymentsProcessedMapper = (
   ~yKey: string,
 ): LineGraphTypes.lineGraphPayload => {
   open LineGraphTypes
-  let categories = getCategories(data, yKey)
-  let data = getLineGraphData(data, xKey)
+  let categories =
+    data
+    ->getArrayFromJson([])
+    ->Array.at(0)
+    ->Option.getOr([]->JSON.Encode.array)
```
Comment (Riddhiagrawal001) [inline src/screens/NewAnalytics/PaymentAnalytics/PaymentsProcessed/PaymentsProcessedUtils.res:15]: Can we use getValueFromArray

File: src/screens/NewAnalytics/PaymentAnalytics/PaymentsSuccessRate/PaymentsSuccessRateUtils.res [hunk 5]
```diff
@@ -19,12 +19,26 @@ let paymentsSuccessRateMapper = (
   ~yKey: string,
 ): LineGraphTypes.lineGraphPayload => {
   open LineGraphTypes
-  let categories = getCategories(data, yKey)
-  let data = getLineGraphData(data, xKey)
+  let categories =
+    data
+    ->getArrayFromJson([])
+    ->Array.at(0)
+    ->Option.getOr([]->JSON.Encode.array)
```
Comment (Riddhiagrawal001) [inline src/screens/NewAnalytics/PaymentAnalytics/PaymentsSuccessRate/PaymentsSuccessRateUtils.res:26]: Can we use getValueFromArray

**Review by Riddhiagrawal001 — APPROVED** [review]

**Review by sagarnaikjuspay — APPROVED** [review]

**Pull request merged**